extensions/notebook/src/test/protocol/notebookUriHandler.test.ts

```diff
@@ -107,9 +114,47 @@ describe('Notebook URI Handler', function (): void {
 				.reply(httpErrorCode);
 
 			await notebookUriHandler.handleUri(notebookUri);
-			
 			sinon.assert.callCount(showErrorMessageSpy, 1);
 			sinon.assert.notCalled(showNotebookDocumentStub);
 		});
 	});
+
+	it('should open notebook when file uri is valid', async function (): Promise<void> {
+		let showQuickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves(Promise.resolve(loc.msgYes) as any);
+		let notebookDir: string = path.join(os.tmpdir(),'notebook');
+		let notebookPath: string = path.join(notebookDir,'hello.ipynb');
+		// Need to remove temp file if it exists
+		if(await exists(notebookDir)){
```

Comment:
And FYI this is an anti-pattern anyways. There's nothing preventing the folder from being created (or deleted) after this check is done which would mean the wrong action could be taken. Generally if you want to do this you just try to delete and then ignore the error if it happens (which would be the case if it didn't exist)